media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala

```diff
@@ -0,0 +1,267 @@
+              case Left(err) => Left(err)
+              case Right(msg) =>
+                logger.debug(s"--> outputting deep archive copy request for ${onlineOutputMessage.nearlineId} for project ${project.id.getOrElse(-1)}")
+                Right(MessageProcessorConverters.contentToMPRV(msg.asJson))
+          }
+        })
+
+
+      case ("check_internal_archive", Some(project)) =>
+        if (_existsInInternalArchive(onlineOutputMessage)) {
+          // media EXISTS in INTERNAL ARCHIVE
+          _removeDeletionPending(onlineOutputMessage)
+          _deleteFromNearline(onlineOutputMessage) match {
+            case Left(err) => Future(Left(err))
+            case Right(msg) =>
+              logger.debug(s"--> deleting ${onlineOutputMessage.nearlineId} for project ${project.id.getOrElse(-1)}")
+              Future(Right(MessageProcessorConverters.contentToMPRV(msg.asJson)))
+          }
+        } else {
+          // media does NOT EXIST in INTERNAL ARCHIVE
+          _storeDeletionPending(onlineOutputMessage)
+          _outputInternalArchiveCopyRequried(onlineOutputMessage) match {
+            case Left(err) => Future(Left(err))
+            case Right(msg) =>
+              logger.debug(s"--> outputting deep archive copy request for ${onlineOutputMessage.nearlineId} for project ${project.id.getOrElse(-1)}")
+              Future(Right(MessageProcessorConverters.contentToMPRV(msg.asJson)))
+          }
+        }
+
+      case ("clear_and_delete", Some(project)) =>
+        // TODO Remove "pending deletion" record if exists
+        _removeDeletionPending(onlineOutputMessage)
+        // TODO Delete media from storage
+        _deleteFromNearline(onlineOutputMessage) match {
+          case Left(err) => Future(Left(err))
+          case Right(msg) =>
+            logger.debug(s"--> outputting deep archive copy request for ${onlineOutputMessage.nearlineId} for project ${project.id.getOrElse(-1)}")
+            Future(Right(MessageProcessorConverters.contentToMPRV(msg.asJson)))
+        }
+
+      case ("just_no", Some(project)) =>
+        logger.warn(s"Project state for removing files from project ${project.id.getOrElse(-1)} is not valid, deep_archive flag is not true!")
+        throw new RuntimeException(s"Project state for removing files from project ${project.id.getOrElse(-1)} is not valid, deep_archive flag is not true!")
+
+    }
+    resultOfAction
+  }
+
+  def bulkGetProjectMetadata(mediaNotRequiredMsg: MultiProjectOnlineOutputMessage) = {
+    mediaNotRequiredMsg.projectIds.map(id => asLookup.getProjectMetadata(id.toString)).sequence
+  }
+
+  def handleNearline(vault: Vault, mediaNotRequiredMsg: MultiProjectOnlineOutputMessage): Future[Either[String, MessageProcessorReturnValue]] = {
+    /*
+    TODO
+     handle seq of projects
+     if any is new or in production, then silentdrop
+     do we need to handle differing statuses other than that, say if differeing in sensitivity?
```

Comment:
- if status is held and the request is to delete from nearline, then also silentdrop
- other statuses are `Completed` and `Killed`, deletion is appropriate in those circu,stances